### PR TITLE
modify shell file of classification for gluon

### PR DESCRIPTION
**Pull request by qingzhouzhen**
I added and checked below function:
1 add loading checkpoint weight
2 make a folder if the path of model-prefix is not exist
3 change learning rate at exact lr-step-epochs

Test my implementation:
`python image_classification.py --dataset=imagenet --train-data=/data/ILSVRC2012_img_train.rec --val-data=/data/ILSVRC2012_img_val.rec --gpus=4 --model=mobilenet0.5 --mode='symbolic' --model-prefix=mobilenet-models/mobilenet --load-epoch=5 --lr-step-epochs='3,6,9' --epochs=10`
load pre_trained weight:
`[12:14:15] src/io/iter_image_recordio_2.cc:153: ImageRecordIOParser2: /data/ILSVRC2012_img_train.rec, use 23 threads for decoding..
[12:14:16] src/io/iter_image_recordio_2.cc:153: ImageRecordIOParser2: /data/ILSVRC2012_img_val.rec, use 23 threads for decoding..
INFO:root:Loaded model mobilenet-models/mobilenet_0005.params
INFO:root:Adjust learning rate to 1.000000e-03 for epoch 5
`

At epoch 6 and epoch 9 learning rate drop as expected:
INFO:root:Epoch[5] Batch [9850] Speed: 1002.63 samples/sec      accuracy=0.414219
INFO:root:Epoch[5] Batch [9900] Speed: 995.15 samples/sec       accuracy=0.426094
INFO:root:Epoch[5] Batch [9950] Speed: 1009.33 samples/sec      accuracy=0.428125
INFO:root:Epoch[5] Batch [10000]        Speed: 1037.10 samples/sec      accuracy=0.423594
INFO:root:Update[10010]: Change learning rate to 1.00000e-04
INFO:root:Epoch[5] Train-accuracy=0.390625
INFO:root:Epoch[5] Time cost=1258.750
INFO:root:Saved checkpoint to "mobilenet-models/mobilenet-0006.params"
INFO:root:Epoch[5] Validation-accuracy=0.417539
INFO:root:Epoch[6] Batch [50]   Speed: 1061.45 samples/sec      accuracy=0.404871
INFO:root:Epoch[6] Batch [100]  Speed: 972.30 samples/sec       accuracy=0.406250

I think there is no need to wright 
`mod.save_params('image-classifier-%s-%d-final.params'%(opt.model, opt.epochs))`,
but for safety, I just motify the syntax error.

### CODE REVIEW

**Comment by zhreshold** [comment]
@szha Now we have it contributed from community!

**Comment by szha** [comment]
@qingzhouzhen would you make sure that the new options you added also work for imperative/hybrid mode?

**Comment by qingzhouzhen** [comment]
This options I add did not work for imperative/hybrid, it only worked for for symbolic mode, currently I have not sufficient time to do this, because I am not quite familiar with gluon and imperative/hybrid mode, Recently I am learning about gluon, I will try to make it, But I am not sure when I will done for  imperative/hybrid mode. @szha

File: example/gluon/image_classification.py [hunk 1]
```diff
@@ -172,16 +223,47 @@ def train(epochs, ctx):
...
+        # Only a limited number of optimizers have 'momentum' property
+        has_momentum = {'sgd', 'dcasgd', 'nag'}
+        if opt.optimizer in has_momentum:
+            optimizer_params['momentum'] = opt.momentum
+
         mod.fit(train_data,
+                begin_epoch=opt.load_epoch if opt.load_epoch else 0,
                 eval_data = val_data,
                 num_epoch=opt.epochs,
                 kvstore=opt.kvstore,
```
Comment (ZiyueHuang) [inline example/gluon/image_classification.py]: I think `kvstore` here should be `kv` you created before, not the string.

Comment (qingzhouzhen) [reply example/gluon/image_classification.py]: yes, I think this is a good suggestion, for it save a instance of 'kv', thx

**Comment by qingzhouzhen** [comment]
And before run, I would have to change `min_random_scale` to 1 in `data.py,` or I would get `input image size smaller than input shape` error

**Comment by zhreshold** [comment]
This is because the input images in record file is resized to like 256x256, and usually people save it as 480 or something.

**Comment by piiswrong** [comment]
closing due to outdated.
@zhreshold Please take over if you think this is still relevant.
